File: src/main/java/rars/venus/settings/editor/EditorSettingsDialog.java
package rars.venus.settings.editor;

import org.jetbrains.annotations.NotNull;
import rars.venus.settings.editor.controllers.EditorSettingsController;
import rars.venus.settings.editor.views.PanelWithTextAreaView;
import rars.venus.settings.editor.views.PickerCardView;

import javax.swing.*;
import java.awt.*;
import java.awt.event.WindowAdapter;
import java.awt.event.WindowEvent;

import static rars.Globals.*;

public final class EditorSettingsDialog extends JDialog {
    public EditorSettingsDialog(
        final @NotNull Frame owner,
        final @NotNull String title,
        final boolean modality
    ) {
        super(owner, title, modality);
        final var pickerCardView = new PickerCardView();
        final var panelWithTextAreaView = new PanelWithTextAreaView(pickerCardView);
        final var treePanel = new TreePanel();
        final var mainPanel = new EditorSettingsPanel(treePanel, panelWithTextAreaView);
        new EditorSettingsController(
            mainPanel,
            this,
            treePanel,
            FONT_SETTINGS,
            EDITOR_THEME_SETTINGS,
            OTHER_SETTINGS
        );
        this.setContentPane(mainPanel);
        this.setDefaultCloseOperation(JDialog.DO_NOTHING_ON_CLOSE);
        this.pack();
        this.addWindowListener(
            new WindowAdapter() {
                @Override
                public void windowClosing(final WindowEvent we) {
                    EditorSettingsDialog.this.setVisible(false);
                    EditorSettingsDialog.this.dispose();
                }
            });
        this.setLocationRelativeTo(owner);
    }
}

File: src/main/java/rars/venus/settings/editor/EditorSettingsDialog.kt
package rars.venus.settings.editor

import rars.settings.AllSettings
import rars.venus.VenusUI
import rars.venus.settings.closeDialog
import rars.venus.settings.editor.controllers.EditorSettingsController
import rars.venus.settings.editor.views.PanelWithTextAreaView
import rars.venus.settings.editor.views.PickerCardView
import rars.venus.util.onWindowClosing
import javax.swing.JDialog

class EditorSettingsDialog(
    mainUI: VenusUI,
    allSettings: AllSettings
) : JDialog(mainUI, "Editor Settings", true) {
    init {
        val (
            boolSettings,
            fontSettings,
            editorThemeSettings,
            highlightingSettings,
            otherSettings,
        ) = allSettings
        val pickerCardView = PickerCardView(
            editorThemeSettings.currentTheme,
            otherSettings.caretBlinkRate,
            otherSettings.editorTabSize
        )
        val panelWithTextAreaView =
            PanelWithTextAreaView(pickerCardView, allSettings)
        val treePanel = TreePanel()
        val mainPanel = EditorSettingsPanel(treePanel, panelWithTextAreaView)
        EditorSettingsController(
            mainPanel,
            this,
            treePanel,
            fontSettings,
            editorThemeSettings,
            otherSettings,
        )
        contentPane = mainPanel
        defaultCloseOperation = DO_NOTHING_ON_CLOSE
        pack()
        onWindowClosing { closeDialog() }
        setLocationRelativeTo(mainUI)
    }
}
